sdk/identity/azure-identity/src/main/java/com/azure/identity/ChainedTokenCredential.java

```diff
@@ -36,29 +38,20 @@
 
     @Override
     public Mono<AccessToken> getToken(TokenRequestContext request) {
-        List<CredentialUnavailableException> exceptions = new ArrayList<>(4);
+        StringBuilder message = new StringBuilder();
         return Flux.fromIterable(credentials)
                    .flatMap(p -> p.getToken(request).onErrorResume(CredentialUnavailableException.class, t -> {
-                       exceptions.add(t);
+                       if (!t.getClass().getSimpleName().equals("CredentialUnavailableException")) {
+                           throw logger.logExceptionAsError(new CredentialUnavailableException(
```

Comment:
This should use Reactor's `Mono.error` instead. Having a throw statement within a reactive stream will cause consumers to need to both catch using try/catch in the code block and add in a Reactor error operator such as `onError`.

Reply:
Reactor actually handles the case where downstream subscribers throw on flatmap. It'll wrap it and propagate a downstream Mono.error. In general though, we should not explicitly throw.
https://github.com/reactive-streams/reactive-streams-jvm#2.13